### PR TITLE
feat(sdk): SDK theming part 2 - bg-error, bg-medium, bg-night, bg-white, border

### PR DESCRIPTION
**Pull request by deniskaber**
Relates to https://github.com/metabase/metabase/issues/42694
Also is a small part of https://www.notion.so/metabase/100-Align-approach-to-Mantine-styles-customization-1066f77c43694c36a6e90a3d2374b731

### Description

Migrate hardcoded colors used in emotion styles:

- bg-error `${color("bg-error")}` -> `var(--mb-color-bg-error)`
- bg-medium `${color("bg-medium")}` -> `var(--mb-color-bg-medium)`, `--color-bg-medium` -> `--mb-color-bg-medium`
- bg-night `${color("bg-night")}` -> `var(--mb-color-bg-night)`
- bg-white `${color("bg-white")}` -> `var(--mb-color-bg-white)`
- border `${color("border")}` -> `var(--mb-color-border)`

### How to verify

1. CI should pass
2. Visual tests should return same results

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/components/private/SdkContentWrapper.tsx [hunk 1]
```diff
@@ -51,6 +51,9 @@ const SdkContentWrapperInner = styled.div<
   --mb-color-text-dark: ${({ theme }) => theme.fn.themeColor("text-dark")};
   --mb-color-text-medium: ${({ theme }) => theme.fn.themeColor("text-medium")};
   --mb-color-text-light: ${({ theme }) => theme.fn.themeColor("text-light")};
+  --mb-color-bg-error: ${() => color("bg-error")};
```
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/components/private/SdkContentWrapper.tsx:54]: @WiNloSt There is `bg-color` in palette.ts which is defined as `"bg-error": "#ED6E6E55"` (same color as `error` with a bit of alpha transparency). I agree that we should add it to `colors.ts` and reference via theme.fn.themeColor("bg-error").

File: enterprise/frontend/src/embedding-sdk/components/private/SdkContentWrapper.tsx [hunk 2]
```diff
@@ -51,6 +51,9 @@ const SdkContentWrapperInner = styled.div<
...
+  --mb-color-bg-medium: ${({ theme }) => theme.fn.themeColor("bg-medium")};
+  --mb-color-bg-night: ${() => color("bg-night")};
```
Comment (heypoom) [inline enterprise/frontend/src/embedding-sdk/components/private/SdkContentWrapper.tsx:56]: Should we also use `theme.fn.themeColor` here and add it to the main palette as well? (Similar to the above notes by Kelvin)

File: enterprise/frontend/src/metabase-enterprise/caching/components/StrategyFormLauncher.styled.tsx [hunk 3]
```diff
@@ -20,9 +20,9 @@ export const PolicyToken = styled(Button)<
   justify-content: center;
   ${({ variant }) =>
     css`
-      border-color: ${color(
-        ["filled", "outline"].includes(variant || "") ? "brand" : "border",
-      )} !important;
+      border-color: ${["filled", "outline"].includes(variant || "")
+        ? color("brand")
```
Comment (heypoom) [inline enterprise/frontend/src/metabase-enterprise/caching/components/StrategyFormLauncher.styled.tsx:24]: shall we use `var(--mb-color-brand)` here too, so it's consistent with `border` below?

File: frontend/src/metabase/collections/components/PinDropZone/PinDropZone.styled.tsx [hunk 4]
```diff
@@ -38,7 +38,9 @@ export const PinDropTargetIndicator = styled.div<PinDropTargetRenderArgs>`
   left: 0;
   right: 0;
   border-left: ${props =>
-    `4px solid ${props.hovered ? color("brand") : color("bg-medium")}`};
+    `4px solid ${
+      props.hovered ? color("brand") : "var(--mb-color-bg-medium)"
```
Comment (heypoom) [inline frontend/src/metabase/collections/components/PinDropZone/PinDropZone.styled.tsx:42]: Perhaps this could be `var(--mb-color-brand)` as well?

**Review by heypoom — COMMENTED** [review]
Looks great! I think the only blocker is the `bg-error` issue Kelvin mentioned above, other than that I think the changes are good. Let me test this locally.

File: frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.styled.tsx [hunk 5]
```diff
@@ -60,7 +60,7 @@ DashboardHeaderButton.defaultProps = {
 };
 
 export const SectionMenuItem = styled(Menu.Item)`
-  background-color: ${darken(color("bg-medium"), 0.1)};
+  background-color: ${() => darken(color("bg-medium"), 0.1)};
```
Comment (heypoom) [inline frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.styled.tsx:63]: nit: I see some places we use `darken("bg-medium", 0.1)` without the additional `color` call, so we can use that instead.

File: frontend/src/metabase/components/ErrorPages/ErrorPages.styled.tsx [hunk 6]
```diff
@@ -1,15 +1,13 @@
...
-
 export const ErrorPageRoot = styled.div<{ bordered?: boolean }>`
   display: flex;
   width: 100%;
   height: 100%;
   flex-direction: column;
   justify-content: center;
   align-items: center;
-  ${({ bordered }) => bordered && `border: 1px solid ${color("border")};`}
+  ${({ bordered }) => bordered && `border: 1px solid var(--mb-color-border);`}
```
Comment (WiNloSt) [inline frontend/src/metabase/components/ErrorPages/ErrorPages.styled.tsx:10]: In a situation like this, I'd put the property in the front so it's obvious what this line is about, then conditionally set the value based on prop instead. If we set the value to `false` the property will removed anyway.

File: frontend/src/metabase/core/components/EditableText/EditableText.styled.tsx [hunk 7]
```diff
@@ -19,14 +19,15 @@ export const EditableTextRoot = styled.div<EditableTextRootProps>`
 
   &:hover,
   &:focus-within {
-    border-color: ${props => (props.isDisabled ? "" : color("border"))};
+    border-color: ${props =>
+      props.isDisabled ? "" : "var(--mb-color-border)"};
```
Comment (WiNloSt) [inline frontend/src/metabase/core/components/EditableText/EditableText.styled.tsx:23]: Do you think this reads a bit better once we swap the condition a bit?
```suggestion
      !props.isDisabled ?? "var(--mb-color-border)"};
```

File: frontend/src/metabase/core/components/SelectButton/SelectButton.styled.tsx [hunk 8]
```diff
@@ -27,7 +27,7 @@ export const SelectButtonRoot = styled.button<SelectButtonRootProps>`
   align-items: center;
   border: 1px solid
     ${({ hasValue, highlighted }) =>
-      hasValue && highlighted ? color("brand") : color("border")};
+      hasValue && highlighted ? color("brand") : "var(--mb-color-border)"};
```
Comment (WiNloSt) [inline frontend/src/metabase/core/components/SelectButton/SelectButton.styled.tsx:30]: should you convert brand color to CSS custom property?

**Review by WiNloSt — COMMENTED** [review]
This is 50% review.

File: frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.styled.tsx (hunk 5 above)
Comment (WiNloSt) [reply frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.styled.tsx:63]: good point. That works, although it won't get the latest value from the theme if we use the color that allows overriding. In that case we need to use. `theme.fn.themeColor(...)` instead. So it will look something like

```ts
darken(theme.fn.themeColor(overriddenColor))
```

File: frontend/src/metabase/palette/components/PaletteResults.tsx [hunk 9]
```diff
@@ -85,7 +83,7 @@ export const PaletteResults = withRouter(
                       isFirst
                         ? undefined
                         : {
-                            borderTop: `1px solid ${color("border")}`,
+                            borderTop: `1px solid var(--mb-color-border)`,
```
Comment (WiNloSt) [inline frontend/src/metabase/palette/components/PaletteResults.tsx:86]: nit: 
```suggestion
                            borderTop: "1px solid var(--mb-color-border)",
```

File: frontend/src/metabase/parameters/components/ParameterWidget/ParameterWidget.styled.ts [hunk 10]
```diff
@@ -13,7 +13,8 @@ export const ParameterFieldSet = styled(FieldSet)<ParameterFieldSetProps>`
   align-items: center;
   transition: opacity 500ms linear;
   border: 2px solid
-    ${props => (props.fieldHasValueOrFocus ? color("brand") : color("border"))};
+    ${props =>
+      props.fieldHasValueOrFocus ? color("brand") : "var(--mb-color-border)"};
```
Comment (WiNloSt) [inline frontend/src/metabase/parameters/components/ParameterWidget/ParameterWidget.styled.ts:17]: should you use `var(--mb-color-brand)` here too?

File: frontend/src/metabase/query_builder/components/NativeQueryEditor/NativeQueryEditor.styled.tsx [hunk 11]
```diff
@@ -155,7 +155,7 @@ export const DragHandleContainer = styled.div`
 export const DragHandle = styled.div`
   width: 100px;
   height: 5px;
-  background-color: ${darken("border", 0.03)};
+  background-color: ${() => darken("border", 0.03)};
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/NativeQueryEditor/NativeQueryEditor.styled.tsx:158]: what's the reason of turning this into a function again?

File: frontend/src/metabase/query_builder/components/notebook/NotebookNativePreview/NotebookNativePreview.tsx [hunk 12]
```diff
@@ -60,7 +60,7 @@ export const NotebookNativePreview = (): JSX.Element => {
   const getErrorMessage = (error: unknown) =>
     typeof error === "string" ? error : undefined;
 
-  const borderStyle = `1px solid ${color("border")}`;
+  const borderStyle = `1px solid var(--mb-color-border)`;
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/notebook/NotebookNativePreview/NotebookNativePreview.tsx:63]: nit: 
```suggestion
  const borderStyle = "1px solid var(--mb-color-border)";
```

File: frontend/src/metabase/query_builder/components/notebook/steps/BreakoutStep/BreakoutStep.unit.spec.tsx [hunk 13]
```diff
@@ -1,6 +1,6 @@
 import userEvent from "@testing-library/user-event";
 
-import { fireEvent, render, screen, getIcon, within } from "__support__/ui";
+import { fireEvent, screen, getIcon, within, render } from "__support__/ui";
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/notebook/steps/BreakoutStep/BreakoutStep.unit.spec.tsx:3]: nit: I would revert this change since it's just the order change.

File: frontend/src/metabase/query_builder/components/view/NativeCodePanel/NativeCodePanel.styled.tsx [hunk 14]
```diff
@@ -14,7 +14,9 @@ export const CodeContainer = styled.pre<CodeContainerProps>`
   box-sizing: border-box;
   margin: 0;
   padding: 1rem;
-  border: 1px solid ${props => color(props.isHighlighted ? "brand" : "border")};
+  border: 1px solid
+    ${props =>
+      props.isHighlighted ? color("brand") : "var(--mb-color-border)"};
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/view/NativeCodePanel/NativeCodePanel.styled.tsx:19]: should you use `--mb-color-brand` here?

File: frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.styled.tsx (hunk 5 above)
Comment (heypoom) [reply frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.styled.tsx:63]: @WiNloSt Given that we have the `setGlobalEmbeddingColors` which re-applies the colors right before rendering, I think it might've already applied the last colors. But yes, using `theme.fn.themeColor` is better as it does not rely on the hack.

```ts
  const themeOverride = useMemo(() => {
    // !! Mutate the global colors object to apply the new colors.
    // This must be done before ThemeProvider calls getThemeOverrides.
    setGlobalEmbeddingColors(theme?.colors, appColors);

    return theme && getEmbeddingThemeOverride(theme);
  }, [appColors, theme]);
```

File: frontend/src/metabase/query_builder/components/view/NativeQueryPreview/NativeQueryPreview.tsx [hunk 15]
```diff
@@ -53,7 +53,7 @@ const CloseButton = ({ onClose }: Pick<NativeQueryPreviewProps, "onClose">) => (
 );
 
 const Divider = () => (
-  <Box mb="lg" style={{ borderTop: `1px solid ${color("border")}` }}></Box>
+  <Box mb="lg" style={{ borderTop: `1px solid var(--mb-color-border)` }}></Box>
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/view/NativeQueryPreview/NativeQueryPreview.tsx:56]: nit: 
```suggestion
  <Box mb="lg" style={{ borderTop: "1px solid var(--mb-color-border)" }}></Box>
```

File: frontend/src/metabase/query_builder/components/view/NewQuestionHeader.jsx [hunk 16]
```diff
@@ -1,14 +1,12 @@
...
-import { color } from "metabase/lib/colors";
-
 import ViewSection, { ViewHeading } from "./ViewSection";
 
 export default function NewQuestionHeader(props) {
   return (
     <ViewSection
       {...props}
-      style={{ borderBottom: `1px solid ${color("border")}` }}
+      style={{ borderBottom: `1px solid var(--mb-color-border)` }}
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/view/NewQuestionHeader.jsx:9]: nit: 
```suggestion
      style={{ borderBottom: "1px solid var(--mb-color-border)" }}
```

File: frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx [hunk 17]
```diff
@@ -133,7 +133,7 @@ export const NotebookContainer = ({
           handle={<Handle />}
           onResizeStop={handleResizeStop}
           style={{
-            borderLeft: `1px solid ${color("border")}`,
+            borderLeft: `1px solid var(--mb-color-border)`,
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/view/View/NotebookContainer/NotebookContainer.tsx:136]: nit: 
```suggestion
            borderLeft: "1px solid var(--mb-color-border)",
```

File: enterprise/frontend/src/metabase-enterprise/caching/components/StrategyFormLauncher.styled.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/metabase-enterprise/caching/components/StrategyFormLauncher.styled.tsx:24]: @WiNloSt I noticed that the brand changes are split into separate PR, so I think this is intentional to make the diffs not too big for review.

https://github.com/metabase/metabase/pull/43598/files

File: frontend/src/metabase/query_builder/components/view/ViewSidebar/ViewSidebar.styled.tsx [hunk 18]
```diff
@@ -17,11 +16,11 @@ export const ViewSidebarAside = styled.aside<{
   ${({ isOpen, side }) =>
     side === "left"
       ? css`
-          border-right: ${isOpen ? `1px solid ${color("border")}` : "none"};
+          border-right: ${isOpen ? `1px solid var(--mb-color-border)` : "none"};
```
Comment (WiNloSt) [inline frontend/src/metabase/query_builder/components/view/ViewSidebar/ViewSidebar.styled.tsx:19]: nit: Use normal quotes

File: frontend/src/metabase/search/components/DropdownSidebarFilter/DropdownSidebarFilter.styled.tsx [hunk 19]
```diff
@@ -15,7 +15,7 @@ export const DropdownFieldSet = styled(FieldSet)<{
     ${({ theme, fieldHasValueOrFocus }) =>
       fieldHasValueOrFocus
         ? theme.fn.themeColor("brand")
-        : theme.fn.themeColor("border")};
+        : "var(--mb-color-border)"};
```
Comment (WiNloSt) [inline frontend/src/metabase/search/components/DropdownSidebarFilter/DropdownSidebarFilter.styled.tsx:18]: should you use `--mb-color-brand` here too?

File: enterprise/frontend/src/metabase-enterprise/caching/components/StrategyFormLauncher.styled.tsx [hunk 20]
```diff
@@ -52,14 +52,16 @@ export const StyledLauncher = styled(Flex)<
   width: 100%;
   ${({ variant }) =>
     css`
-      border-color: ${color(
-        ["filled", "outline"].includes(variant || "") ? "brand" : "border",
-      )} !important;
+      border-color: ${["filled", "outline"].includes(variant || "")
+        ? color("brand")
```
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/metabase-enterprise/caching/components/StrategyFormLauncher.styled.tsx:56]: addressed in #43598

File: frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.styled.tsx [hunk 21]
```diff
@@ -115,7 +115,7 @@ export const getCheckboxOverrides = (): MantineThemeOverride["components"] => ({
           },
         },
         labelWrapper: {
-          psoition: "relative",
+          position: "relative",
```
Comment (WiNloSt) [inline frontend/src/metabase/ui/components/inputs/Checkbox/Checkbox.styled.tsx:118]: since the previous change has a typo, have you looked at the component if it behaves differently now?

File: frontend/src/metabase/admin/permissions/components/PermissionsTable/PermissionsTable.styled.tsx [hunk 22]
```diff
@@ -80,7 +82,7 @@ export const EntityNameLink = styled(Link)`
   display: inline;
   font-weight: 700;
   text-decoration: underline;
-  color: ${color("admin-navbar")};
+  color: ${() => color("admin-navbar")};
```
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/admin/permissions/components/PermissionsTable/PermissionsTable.styled.tsx:85]: @WiNloSt I think it's because of this logic I mentioned in the above comments about `setGlobalEmbeddingColors`, which sets the global colors object. If `color` is not a function call it does not re-apply. See `SdkThemeProvider.ts`

But yes if we add `admin-navbar` to Mantine theme colors then we don't need the () => ...` hack

```ts
  const themeOverride = useMemo(() => {
    // !! Mutate the global colors object to apply the new colors.
    // This must be done before ThemeProvider calls getThemeOverrides.
    setGlobalEmbeddingColors(theme?.colors, appColors);

    return theme && getEmbeddingThemeOverride(theme);
  }, [appColors, theme]);
```

File: frontend/src/metabase/collections/components/PinDropZone/PinDropZone.styled.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/collections/components/PinDropZone/PinDropZone.styled.tsx:42]: This is addressed in #43598

File: frontend/src/metabase/visualizations/components/TableSimple/TableSimple.styled.tsx [hunk 23]
```diff
@@ -42,14 +42,14 @@ export const Table = styled.table`
   ${standardTableStyleReset}
 
   tr {
-    border-bottom: 1px solid ${alpha(color("border"), 0.3)};
+    border-bottom: 1px solid ${() => alpha(color("border"), 0.3)};
```
Comment (WiNloSt) [inline frontend/src/metabase/visualizations/components/TableSimple/TableSimple.styled.tsx:45]: this is another place where we convert the expression to a function. I'd like to know what was the reason for that.

File: frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/BaseBucketPickerPopover.styled.tsx [hunk 24]
```diff
@@ -28,7 +28,7 @@ export const TriggerButton = styled.button`
 
   color: ${alpha(color("white"), 0.5)};
   font-weight: 700;
-  border-left: 2px solid ${alpha(color("border"), 0.1)};
+  border-left: 2px solid ${() => alpha(color("border"), 0.1)};
```
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/common/components/QueryColumnPicker/BucketPickerPopover/BaseBucketPickerPopover.styled.tsx:31]: explained in https://github.com/metabase/metabase/pull/43512#discussion_r1627510527, but yes we can use `theme.fn.themeColor` instead of this

File: frontend/src/metabase/visualizations/lib/save-dashboard-pdf.ts [hunk 25]
```diff
@@ -21,7 +19,7 @@ export const saveDashboardPdf = async (
       node.classList.add(SAVING_DOM_IMAGE_CLASS);
       const title = doc.createElement("h2") as HTMLElement;
       title.innerHTML = dashboardName;
-      title.style["borderBottom"] = `1px solid ${color("border")}`;
+      title.style["borderBottom"] = `1px solid var(--mb-color-border)`;
```
Comment (WiNloSt) [inline frontend/src/metabase/visualizations/lib/save-dashboard-pdf.ts:22]: nit: 
```suggestion
      title.style["borderBottom"] = "1px solid var(--mb-color-border)";
```

**Review by WiNloSt — COMMENTED** [review]
Awesome work. I think there are some minor feedback that should be addressed first, but it's almost ready to be merged.

File: frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.styled.tsx (hunk 5 above)
Comment (WiNloSt) [reply frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeader.styled.tsx:63]: that's new to me. Good to know we have this fallback at least. 👍

File: enterprise/frontend/src/metabase-enterprise/caching/components/StrategyFormLauncher.styled.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (WiNloSt) [reply enterprise/frontend/src/metabase-enterprise/caching/components/StrategyFormLauncher.styled.tsx:24]: That makes sense.

File: frontend/src/metabase/admin/permissions/components/PermissionsTable/PermissionsTable.styled.tsx (hunk 22 above)
[the remark this answers is not in the document]
Comment (WiNloSt) [reply frontend/src/metabase/admin/permissions/components/PermissionsTable/PermissionsTable.styled.tsx:85]: Ok, that makes sense. I feel that eventually there should be only one way of dealing with things. By introducing that `setGlobalEmbeddingColors` function, we assume we'll keep using `color` along side `theme.fn.themeColor` which is a bit confusing to me.

Do you envision we keep using both `color` and `theme.fn.themeColor`? I feel like we should just use CSS custom properties e.g. --mb-color-brand and `theme.fn.themeColor` when we need to use util like `darken` and `lighten`.

What do you thinK?

[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/admin/permissions/components/PermissionsTable/PermissionsTable.styled.tsx:85]: @WiNloSt Yep, I think we should advocate for CSS variables first when possible, then `theme.fn.themeColor` secondary. Let's try to not use `color(...)` when we can.

That `setGlobalEmbeddingColors` is intended to be a temporary solution to aid us in migration, not a permanent one.

[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/admin/permissions/components/PermissionsTable/PermissionsTable.styled.tsx:85]: Agree! With the latest discussions on Mantine colors in mind, `() => color("admin-navbar")` is good.

File: frontend/src/metabase/components/ErrorPages/ErrorPages.styled.tsx (hunk 6 above)
Comment (WiNloSt) [reply frontend/src/metabase/components/ErrorPages/ErrorPages.styled.tsx:10]: I know. That's why reviews exist. You can incorporate changes as you see fit.

**Review by heypoom — APPROVED** [review]
Reviewing the [latest diffs](https://github.com/metabase/metabase/pull/43512/files/6cae816d9d038b9cb13bbbd61545cacaaaaf6500..294adf0fa444d07ccf10c2c6ed96decddce21ce3) the changes looks good, thanks Denis!

File: frontend/src/metabase/visualizations/components/MiniBar.jsx [hunk 26]
```diff
@@ -81,7 +81,7 @@ const MiniBar = ({ value, extent: [min, max], options }) => {
               left: "50%",
               top: 0,
               bottom: 0,
-              borderLeft: `1px solid white`,
+              borderLeft: "1px solid var(--mb-color-white)",
```
Comment (WiNloSt) [inline frontend/src/metabase/visualizations/components/MiniBar.jsx:84]: I don't think we have `--mb-color-white` yet.

Comment (WiNloSt) [reply frontend/src/metabase/visualizations/components/MiniBar.jsx:84]: I guess you should use `color("white")` instead.

Comment (heypoom) [reply frontend/src/metabase/visualizations/components/MiniBar.jsx:84]: Nice catch!

**Review by WiNloSt — APPROVED** [review]
I haven't crossed check with all the comments I've left, but from the change itself it looks good.
